tests/data_files/Makefile

```diff
@@ -1054,7 +1054,10 @@ server1.v1.der.openssl: server1.v1.crt.openssl
 crl.pem: $(test_ca_crt) $(test_ca_key_file_rsa) $(test_ca_config_file)
 	$(OPENSSL) ca -gencrl -batch -cert $(test_ca_crt) -keyfile $(test_ca_key_file_rsa) -key $(test_ca_pwd_rsa) -config $(test_ca_server1_config_file) -md sha1 -crldays 3653 -out $@
 
-server1_all: crl.pem server1.crt server1.noauthid.crt server1.crt.openssl server1.v1.crt server1.v1.crt.openssl server1.key_usage.crt server1.key_usage_noauthid.crt server1.key_usage.crt.openssl server1.cert_type.crt server1.cert_type_noauthid.crt server1.cert_type.crt.openssl server1.der server1.der.openssl server1.v1.der server1.v1.der.openssl server1.key_usage.der server1.key_usage.der.openssl server1.cert_type.der server1.cert_type.der.openssl
+crl-futureRevocationDate.pem: $(test_ca_crt) $(test_ca_key_file_rsa) $(test_ca_config_file)
+	$(FAKETIME) '3000-01-01' $(OPENSSL) ca -gencrl -batch -cert $(test_ca_crt) -keyfile $(test_ca_key_file_rsa) -key $(test_ca_pwd_rsa) -config $(test_ca_server1_config_file) -md sha1 -crldays 3653 -out $@
```

Comment:
This looked right to me, but the result is not what we want: it sets the certificates' revocation dates to 2011-02-12 12:44:07 GMT. The current time (changed with faketime) and `-crldays` only influence the update times for the CRL, not the revocation date of the entries. The revocation date is coming from the start of the validity date of `test-ca.crt`. I guess it makes sense that the revocation of a certificate would be back-dated to the earliest time when it might have been generated. But it's not what we want here.

I don't know if there's a way to generate the test data that we want here with an OpenSSL command line. If there isn't, binary patching is acceptable as a last resort, as long as it's clearly explained in a comment.

Reply:
I think you can change the revocation date in `index.txt`.